### ISSUE TITLE
Bug: jsx-closing-bracket-location autofix removes whitespace character

### ISSUE THREAD

**Issue by TSMMark** [yannickcr/eslint-plugin-react#1501]
```jsx
<div>
  foo </div>
```

Renders a space in HTML: `<div>foo </div>`

After fix is applied, it becomes:

```jsx
<div>
  foo 
</div>
```

Note that there IS a space after `foo` in the output, however this space will typically be linted off by `no-trailing-spaces` rule, and even if it's not linted out of your source code, it will be ignored by React's JSX parser, meaning the output is different 100% of the time.

tl;dr

`<div>foo </div>` becomes `<div>foo</div>`

**Comment by ljharb** [comment]
In that case it seems like the autofix would need to be:
```jsx
<div>
  {'foo '}
</div>
```

but i'm not sure it's practical to preserve the space in situations like this; the best practice when a space is important is to put it inside a string inside curlies, for reasons like this.

**Comment by TSMMark** [comment]
@ljharb I totally agree that if your whitespace is important you should contain it in curlies. But can we really justify the autofixer breaking code, even if the code is "bad"? If yes, what do you say I apply that same logic to #1497 when handling whitespace? (it's driving me nuts trying to handle it lol)

**Comment by ljharb** [comment]
No, we can't justify that - you're correct.

**Comment by Moong0122** [comment]
I will take a look this issue
